File: src/main/java/org/hwabeag/playertime/commands/MainCommand.java
package org.hwabeag.playertime.commands;

import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.hwabeag.playertime.config.ConfigManager;
import org.hwabeag.playertime.inventorys.TopPlayerTimeGUI;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class MainCommand implements CommandExecutor {

    FileConfiguration PlayerTimeConfig = ConfigManager.getConfig("playertime");
    String Prefix = ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(PlayerTimeConfig.getString("playertime.prefix")));

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (!(sender instanceof Player p)) {
            Bukkit.getConsoleSender().sendMessage(Prefix + " 인게임에서만 사용이 가능합니다.");
            if (args.length == 0) {
                Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 정보 [닉네임] - 나의 정보 혹 다른 유저의 정보를 확인합니다.");
                Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
                Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 리로드 - 콘피그 정보를 리로드 합니다.");
                return true;
            }
            if (args[0].equalsIgnoreCase("정보")) {
                if (args.length == 1) {
                    Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 정보 [닉네임] - 나의 정보 혹 다른 유저의 정보를 확인합니다.");
                    return true;
                }
                if (args.length == 2) {
                    Player target = Bukkit.getServer().getPlayerExact(args[1]);
                    String Name = Objects.requireNonNull(target).getName();
                    if (PlayerTimeConfig.getString("접속시간." + Name) == null) {
                        Bukkit.getConsoleSender().sendMessage(Prefix + " 해당 닉네임의 정보는 존재하지 않습니다.");
                        return true;
                    }
                    String TimeNumber = PlayerTimeConfig.getString("접속시간." + Name);
                    Bukkit.getConsoleSender().sendMessage(Prefix + " 해당 유저의 플레이 타임은 " + TimeNumber + " 분 입니다.");
                    return true;
                }
            }
            if (args[0].equalsIgnoreCase("설정")) {
                if (args.length == 1) {
                    Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
                    return true;
                }
                if (args.length == 2) {
                    Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
                    return true;
                }
                if (Bukkit.getServer().getPlayerExact(args[1]) != null) {
                    Player target = Bukkit.getServer().getPlayerExact(args[1]);
                    String Name = Objects.requireNonNull(target).getName();
                    if (PlayerTimeConfig.getString("접속시간." + Name) != null) {
                        PlayerTimeConfig.set("접속시간." + Name, args[2]);
                        Bukkit.getConsoleSender().sendMessage(Prefix + " " + Name + " 님의 시간을 " + args[2] + "분으로 설정했습니다.");
                        ConfigManager.saveConfigs();
                    } else {
                        PlayerTimeConfig.addDefault("접속시간." + Name, 0);
                        Bukkit.getConsoleSender().sendMessage(Prefix + " " + Name + " 님의 플레이타임 정보를 생성했습니다.");
                        ConfigManager.saveConfigs();
                    }
                } else {
                    Bukkit.getConsoleSender().sendMessage(Prefix + " " + args[1] + " 닉네임의 유저가 존재하지 않습니다.");
                }
                return true;
            }
            if (args[0].equalsIgnoreCase("리로드")) {
                ConfigManager.reloadConfigs();
                Bukkit.getConsoleSender().sendMessage(Prefix + " 플레이타임 콘피그가 리로드 되었습니다.");
                return true;
            }
            Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 정보 [닉네임] - 나의 정보 혹 다른 유저의 정보를 확인합니다.");
            Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
            Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 리로드 - 콘피그 정보를 리로드 합니다.");
            return false;
        }
        if (args.length == 0) {
            p.sendMessage(Prefix + " /플레이타임 정보 [닉네임] - 나의 정보 혹 다른 유저의 정보를 확인합니다.");
            p.sendMessage(Prefix + " /플레이타임 순위 - 접속시간 순위를 확인합니다.");
            if (p.isOp()) {
                p.sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
                p.sendMessage(Prefix + " /플레이타임 리로드 - 콘피그 정보를 리로드 합니다.");
            }
            return true;
        }
        if (args[0].equalsIgnoreCase("순위")) {
            if (args.length == 1) {
                TopPlayerTimeGUI inv = new TopPlayerTimeGUI();
                inv.open(p);
                return true;
            }
        }
        /* String TimeNumber = PlaceholderAPI.setPlaceholders(p, "%playertime_get%"); */
        // p.sendMessage(Prefix + " 당신의 플레이 타임은 " + TimeNumber + " 분 입니다."); */
        if (args[0].equalsIgnoreCase("정보")) {
            if (args.length == 1) {
                String Name = Objects.requireNonNull(p).getName();
                if (PlayerTimeConfig.getString("접속시간." + Name) == null) {
                    p.sendMessage(Prefix + " 당신의 정보가 존재하지 않습니다.");
                    return true;
                }
                String TimeNumber = PlayerTimeConfig.getString("접속시간." + Name);
                p.sendMessage(Prefix + " 당신의 플레이 타임은 " + TimeNumber + " 분 입니다.");
                return true;
            }
            if (args.length == 2) {
                Player target = Bukkit.getServer().getPlayerExact(args[1]);
                String Name = Objects.requireNonNull(target).getName();
                if (PlayerTimeConfig.getString("접속시간." + Name) == null) {
                    p.sendMessage(Prefix + " 해당 닉네임의 정보는 존재하지 않습니다.");
                    return true;
                }
                String TimeNumber = PlayerTimeConfig.getString("접속시간." + Name);
                p.sendMessage(Prefix + " 해당 유저의 플레이 타임은 " + TimeNumber + " 분 입니다.");
                return true;
            }
        }
        if (p.isOp()) {
            if (args[0].equalsIgnoreCase("설정")) {
                if (args.length == 1) {
                    p.sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
                    return true;
                }
                if (args.length == 2) {
                    p.sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
                    return true;
                }
                if (Bukkit.getServer().getPlayerExact(args[1]) != null) {
                    Player target = Bukkit.getServer().getPlayerExact(args[1]);
                    String Name = Objects.requireNonNull(target).getName();
                    if (PlayerTimeConfig.getString("접속시간." + Name) != null) {
                        PlayerTimeConfig.set("접속시간." + Name, args[2]);
                        p.sendMessage(Prefix + " " + Name + " 님의 시간을 " + args[2] + "분으로 설정했습니다.");
                        ConfigManager.saveConfigs();
                    } else {
                        PlayerTimeConfig.addDefault("접속시간." + Name, 0);
                        p.sendMessage(Prefix + " " + Name + " 님의 플레이타임 정보를 생성했습니다.");
                        ConfigManager.saveConfigs();
                    }
                } else {
                    p.sendMessage(Prefix + " " + args[1] + " 닉네임의 유저가 존재하지 않습니다.");
                }
                return true;
            }
            if (args[0].equalsIgnoreCase("리로드")) {
                ConfigManager.reloadConfigs();
                p.sendMessage(Prefix + " 플레이타임 콘피그가 리로드 되었습니다.");
                return true;
            }
            p.sendMessage(Prefix + " /플레이타임 정보 [닉네임] - 나의 정보 혹 다른 유저의 정보를 확인합니다.");
            if (p.isOp()) {
                p.sendMessage(Prefix + " /플레이타임 설정 [닉네임] [분] - 유저의 정보를 설정합니다.");
                p.sendMessage(Prefix + " /플레이타임 리로드 - 콘피그 정보를 리로드 합니다.");
                Bukkit.getConsoleSender().sendMessage(Prefix + " /플레이타임 papi - papi 테스트 명령어");
            }
        } else {
            p.sendMessage(Prefix + " 당신은 권한이 없습니다.");
        }
        return true;
    }
}

File: src/main/kotlin/org/hwabeag/playertime/commands/MainCommand.kt
package org.hwabeag.playertime.commands

import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.hwabeag.playertime.PlayerTime
import org.hwabeag.playertime.inventory.TopPlayerTimeGui

class MainCommand(private val plugin: PlayerTime) : CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        val prefix = plugin.configManager.prefix

        if (args.isEmpty()) {
            sender.sendMessage("$prefix /플레이타임 정보 [닉네임]")
            sender.sendMessage("$prefix /플레이타임 순위")
            if (sender.isOp) {
                sender.sendMessage("$prefix /플레이타임 설정 <닉네임> <분>")
                sender.sendMessage("$prefix /플레이타임 지급 <닉네임> <분>")
                sender.sendMessage("$prefix /플레이타임 리로드")
            }
            return true
        }

        return when (args[0].lowercase()) {
            "정보" -> handleInfo(sender, args)
            "순위" -> handleTop(sender)
            "설정" -> handleSet(sender, args)
            "지급" -> handleAdd(sender, args)
            "리로드" -> handleReload(sender)
            else -> {
                sender.sendMessage("$prefix 알 수 없는 하위 명령어입니다.")
                true
            }
        }
    }

    private fun handleInfo(sender: CommandSender, args: Array<out String>): Boolean {
        val prefix = plugin.configManager.prefix
        val targetName = if (args.size >= 2) args[1] else (sender as? Player)?.name

        if (targetName == null) {
            sender.sendMessage("$prefix 콘솔에서는 닉네임을 함께 입력해야 합니다.")
            return true
        }

        val minutes = plugin.storage.getMinutesByName(targetName)
        if (minutes == null) {
            sender.sendMessage("$prefix $targetName 님의 데이터가 없습니다.")
            return true
        }

        sender.sendMessage("$prefix $targetName 님의 플레이타임은 ${minutes}분입니다.")
        return true
    }

    private fun handleTop(sender: CommandSender): Boolean {
        val prefix = plugin.configManager.prefix
        if (sender !is Player) {
            sender.sendMessage("$prefix 콘솔에서는 사용할 수 없습니다.")
            return true
        }

        TopPlayerTimeGui(plugin).open(sender)
        return true
    }

    private fun handleSet(sender: CommandSender, args: Array<out String>): Boolean {
        val prefix = plugin.configManager.prefix
        if (!sender.isOp) {
            sender.sendMessage("$prefix 권한이 없습니다.")
            return true
        }

        if (args.size < 3) {
            sender.sendMessage("$prefix 사용법: /플레이타임 설정 <닉네임> <분>")
            return true
        }

        val name = args[1]
        val minutes = args[2].toIntOrNull()
        if (minutes == null || minutes < 0) {
            sender.sendMessage("$prefix 분 값은 0 이상의 숫자여야 합니다.")
            return true
        }

        val online = Bukkit.getPlayerExact(name)
        if (online != null) {
            plugin.storage.ensurePlayer(online.uniqueId.toString(), online.name)
        }

        val changed = plugin.storage.setMinutesByName(name, minutes)
        if (!changed) {
            sender.sendMessage("$prefix $name 님의 데이터가 없어 설정하지 못했습니다. 한 번 이상 접속한 플레이어인지 확인하세요.")
            return true
        }

        sender.sendMessage("$prefix $name 님의 플레이타임을 ${minutes}분으로 설정했습니다.")
        return true
    }

    private fun handleAdd(sender: CommandSender, args: Array<out String>): Boolean {
        val prefix = plugin.configManager.prefix
        if (!sender.isOp) {
            sender.sendMessage("$prefix 권한이 없습니다.")
            return true
        }

        if (args.size < 3) {
            sender.sendMessage("$prefix 사용법: /플레이타임 지급 <닉네임> <분>")
            return true
        }

        val name = args[1]
        val minutes = args[2].toIntOrNull()
        if (minutes == null || minutes <= 0) {
            sender.sendMessage("$prefix 분 값은 1 이상의 숫자여야 합니다.")
            return true
        }

        val online = Bukkit.getPlayerExact(name)
        if (online != null) {
            plugin.storage.ensurePlayer(online.uniqueId.toString(), online.name)
        }

        val total = plugin.storage.addMinutesByName(name, minutes)
        if (total == null) {
            sender.sendMessage("$prefix $name 님의 데이터가 없어 지급하지 못했습니다. 한 번 이상 접속한 플레이어인지 확인하세요.")
            return true
        }

        sender.sendMessage("$prefix $name 님에게 ${minutes}분을 지급했습니다. 현재 ${total}분입니다.")
        return true
    }

    private fun handleReload(sender: CommandSender): Boolean {
        val prefix = plugin.configManager.prefix
        if (!sender.isOp) {
            sender.sendMessage("$prefix 권한이 없습니다.")
            return true
        }

        plugin.configManager.reload()
        sender.sendMessage("$prefix 설정을 다시 불러왔습니다.")
        return true
    }
}
